actions/download_samplesheet_clarity.py

```diff
@@ -0,0 +1,53 @@
+from st2actions.runners.pythonrunner import Action
+
+from clarity_ext.context import ExtensionContext
+from genologics.lims import *
+from genologics.config import BASEURI, USERNAME, PASSWORD
+
+import os
+from tempfile import mkdtemp
+
+class DownloadSamplesheetClarity(Action):
+    """
+    Downloads a samplesheet from Clarity LIMS (instead of Hermes)
+    keyed on the flowcell id.
+    """
+
+    def get_processes(self, containers):
+        for container in containers:
+            placement = container.get_placements()
+            artifact = placement[placement.keys()[0]]
+            yield artifact.parent_process.id
+
+    def highest_id(self, process_ids):
+        highest_proc_number = 0
+        highest_proc_type = ''
+        for id in process_ids:
```

Comment:
Since `id` is a built in function it might be preferable to use an other name here. https://docs.python.org/2/library/functions.html#id